asmcomp/linearize.ml

```diff
@@ -308,7 +234,7 @@ let rec linear i n =
       n3
   | Iexit nfail ->
       let lbl, t = find_exit_label_try_depth nfail in
-      assert (i.Mach.next.desc = Iend);
+      assert (i.Mach.next.desc = Mach.Iend);
```

Comment:
Is this strictly necessary?

Reply:
Without it, I get
``` Error (warning 18): this type-based constructor disambiguation is not principal. ```
Is there another way to fix it?

Reply:
Oh, it's because of the loss of the `open Mach` at the top - sorry for the noise!